Highway.cpp

```diff
@@ -13,7 +18,22 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    //assert(false);
+
+    if( Car* car = dynamic_cast<Car*>(v) )
+    {
+        car->closeWindows();
+    }
+
+    if ( Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v) )
+    {
+        motorcycle->lanesplitAndRace(speedLimit + 50);
+    }
+
+    if ( SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v) )
+    {
+        semiTruck->dumpTheCargo();
+    }
```

Comment:
use `else if`.  no point checking if a Car is a Motorcycle if you already figured out it's a Car
use `auto* thing = dynamic_cast<>`
- [ ] revise